x-pack/plugins/index_lifecycle_management/server/routes/api/templates/register_add_policy_route.ts

```diff
@@ -5,45 +5,78 @@
  */
 
 import { merge } from 'lodash';
-import { schema } from '@kbn/config-schema';
+import { schema, TypeOf } from '@kbn/config-schema';
 import { LegacyAPICaller } from 'src/core/server';
+import { i18n } from '@kbn/i18n';
 
+import { TemplateFromEs, TemplateSerialized } from '../../../../../index_management/common/types';
 import { LegacyTemplateSerialized } from '../../../../../index_management/server';
 import { RouteDependencies } from '../../../types';
 import { addBasePath } from '../../../services';
 
-async function getIndexTemplate(
+async function getLegacyIndexTemplate(
   callAsCurrentUser: LegacyAPICaller,
   templateName: string
-): Promise<LegacyTemplateSerialized> {
+): Promise<LegacyTemplateSerialized | undefined> {
   const response = await callAsCurrentUser('indices.getTemplate', { name: templateName });
   return response[templateName];
 }
 
+async function getIndexTemplate(
+  callAsCurrentUser: LegacyAPICaller,
```

Comment:
It seems that the `LegacyAPICaller` is deprecated, maybe we should find out which interface we should use here?

<img width="352" alt="Screenshot 2020-09-17 at 11 24 47" src="https://user-images.githubusercontent.com/2854616/93452285-8834af80-f8d8-11ea-92f2-5aad5e59a880.png">

Reply:
Yes, I noticed the deprecation warning in several places in this plugin (probably in all other plugins as well). Created an issue for the ILM server code [#77906](https://github.com/elastic/kibana/issues/77906)